sklearn/metrics/ranking.py

```diff
@@ -857,3 +858,278 @@ def label_ranking_loss(y_true, y_score, sample_weight=None):
     loss[np.logical_or(n_positives == 0, n_positives == n_labels)] = 0.
 
     return np.average(loss, weights=sample_weight)
+
+
+def _dcg_sample_scores(y_true, y_score, k=None, log_basis=2):
+    """Compute Discounted Cumulative Gain.
+
+    Sum the true scores ranked in the order induced by the predicted scores,
+    after applying a logarithmic discount.
+
+    This ranking metric yields a high value if true labels are ranked high by
+    ``y_score``.
+
+    Parameters
+    ----------
+    y_true : array, shape = [n_samples, n_labels], dtype = int or float
+        True targets of multilabel classification, or true scores of entities
+        to be ranked.
+
+    y_score : array, shape = [n_samples, n_labels], dtype = float
+        Target scores, can either be probability estimates, confidence values,
+        or non-thresholded measure of decisions (as returned by
+        "decision_function" on some classifiers).
+
+    k : int, optional (default=None)
+        Only consider the highest k scores in the ranking. If None, use all
+        outputs.
+
+    log_basis : float, optional (default=2)
+        Basis of the logarithm used for the discount. A low value means a
+        sharper discount (top results are more important).
+
+    Returns
+    -------
+    discounted_cumulative_gain : array, shape = [n_samples,], dtype = float
+        The DCG score for each sample.
+
+    See also
+    --------
+    ndcg_score :
+        The Discounted Cumulative Gain divided by the Ideal Discounted
+        Cumulative Gain (the DCG obtained for a perfect ranking), in order to
+        have a score between 0 and 1.
+
+    """
+    check_array(y_true, ensure_2d=False)
+    check_array(y_score, ensure_2d=False)
+    check_consistent_length(y_true, y_score)
+    y_type = type_of_target(y_true)
+    if y_type not in ("multilabel-indicator", "continuous-multioutput",
```

Comment:
actually, can we make the common checking in the public function and having the barely minimum check in the private one.

Reply:
done. the incovenient is that the check must now be repeated for dcg and ndcg